### PR TITLE
fix: bump libsession to allow createdAt to be given during migration

### PR DESCRIPTION
**Pull request by Bilb**

### CODE REVIEW

File: ts/node/sql.ts [hunk 1]
```diff
@@ -972,11 +971,13 @@ function removeAllMessagesInConversation(
   if (!conversationId) {
     return;
   }
+  const inst = assertGlobalInstanceOrInstance(instance);
 
-  // Our node interface doesn't seem to allow you to replace one single ? with an array
-  assertGlobalInstanceOrInstance(instance)
-    .prepare(`DELETE FROM ${MESSAGES_TABLE} WHERE conversationId = $conversationId`)
-    .run({ conversationId });
+  inst.transaction(() => {
```
Comment (yougotwill) [inline ts/node/sql.ts:976]: what is the benefit of wrapping this prepare statement in a transaction?

**Review by yougotwill — APPROVED** [review]
Just a couple of questions but LGTM 🚀

File: ts/node/migration/sessionMigrations.ts [hunk 2]
```diff
@@ -1863,5 +1887,8 @@ export function updateSessionSchema(db: BetterSqlite3.Database) {
   for (let index = 0, max = LOKI_SCHEMA_VERSIONS.length; index < max; index += 1) {
     const runSchemaUpdate = LOKI_SCHEMA_VERSIONS[index];
     runSchemaUpdate(lokiSchemaVersion, db);
+    if (index > lokiSchemaVersion) {
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/node/migration/sessionMigrations.ts:1890]: that's right

File: ts/node/sql.ts (hunk 1 above)
Comment (Bilb) [reply ts/node/sql.ts:976]: This was a test to run the triggers called while deleting messages only once, in an attempt to batch them at the database level. But I don't think it helps, what helped was the change of id to rowId in the fts virtual table

File: ts/node/migration/sessionMigrations.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (yougotwill) [reply ts/node/migration/sessionMigrations.ts:1890]: Cool

File: ts/node/sql.ts (hunk 1 above)
Comment (yougotwill) [reply ts/node/sql.ts:976]: 👍

**Review by yougotwill — APPROVED** [review]
Very nice work dude! I like how we are using selectors and hooks more. I'll try and do the same. It looks much cleaner to me 😄

File: ts/components/search/MessageSearchResults.tsx [hunk 3]
```diff
@@ -172,7 +172,7 @@ const StyledTimestampContaimer = styled.div`
   color: var(--conversation-tab-text-color);
 `;
 
-export const MessageSearchResult = (props: MessageResultProps) => {
+export const MessageSearchResult = (props: MessageResultProps & { style: CSSProperties }) => {
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/components/search/MessageSearchResults.tsx:175]: fixed

File: ts/components/search/SearchResults.tsx [hunk 4]
```diff
@@ -35,37 +38,58 @@ const NoResults = styled.div`
...
-export const SearchResults = (props: SearchResultsProps) => {
-  const { contactsAndGroupsIds, messages, searchTerm } = props;
+const SectionHeader = ({ title, style }: { title: string; style: CSSProperties }) => {
+  return <StyledSeparatorSection style={style}>{title}</StyledSeparatorSection>;
+};
 
-  const haveContactsAndGroup = Boolean(contactsAndGroupsIds?.length);
-  const haveMessages = Boolean(messages?.length);
-  const noResults = !haveContactsAndGroup && !haveMessages;
+function isContact(item: SearchResultsMergedListItem): item is { contactConvoId: string } {
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/components/search/SearchResults.tsx:45]: this is called as a type guard 
check the "Custom type guard with predicate"
of https://blog.logrocket.com/how-to-use-type-guards-typescript/

Fixed contact and groups count

[the remark this answers is not in the document]
Comment (Bilb) [reply ts/components/search/SearchResults.tsx:45]: also made it to have the `: $count$` localized

File: ts/state/selectors/search.ts [hunk 5]
```diff
@@ -41,13 +42,43 @@ const getSearchResults = createSelector(
...
+
 export const getSearchResultsContactOnly = createSelector([getSearchResults], searchState => {
   return searchState.contactsAndGroups.filter(m => m.isPrivate).map(m => m.id);
 });
+
+/**
+ *
+ * When type is string, we render a sectionHeader.
+ * When type just has a conversationId field, we render a ConversationListItem.
+ * When type is MessageResultProps we render a MessageSearchResult
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/state/selectors/search.ts:68]: yeah, that's the only way I can think of to make sections and render different type of items in the virtualized list.
This is probably the recommended way. I remember having to do it like that a while ago already.

[the remark this answers is not in the document]
Comment (yougotwill) [reply ts/state/selectors/search.ts:68]: All good

**Review by yougotwill — APPROVED** [review]
LGTM

**Pull request merged**